core/src/main/java/com/kin/ecosystem/core/data/auth/AuthRepository.java

```diff
@@ -90,7 +90,7 @@ public String getUserID() {
 
 	@Override
 	public String getEcosystemUserID() {
-		return localData.getEcosystemUserID();
+		return cachedAuthToken == null ? localData.getEcosystemUserID() : cachedAuthToken.getEcosystemUserID();
```

Comment:
cached auth token complicate the logic localData.getEcosystemUserID() should handle caching internally if needed

Reply:
no, it's the repo responsibility and it's not complicated.

Reply:
OK